### PR TITLE
Fix issue with loading of nncf part of config from checkpoint metainfo

### PR DESCRIPTION
**Pull request by LeonidBeynenson**
This part should be written to a file and then loaded from it, but
bufferization may prevent loading of the data while the file descriptor
is not closed.

### CODE REVIEW

File: mmdet/integration/nncf/compression.py [hunk 1]
```diff
@@ -78,12 +78,24 @@ def get_nncf_config_from_meta(path):
             'get_nncf_config_from_meta should be run for NNCF-compressed checkpoints only'
 
     config_text = meta['config']
-    with tempfile.NamedTemporaryFile(prefix='config_', suffix='.py', mode='w') as f_tmp:
+
+    with tempfile.NamedTemporaryFile(prefix='config_', suffix='.py',
+                                     mode='w', delete=False) as f_tmp:
         f_tmp.write(config_text)
-        cfg = mmcv.Config.fromfile(f_tmp.name)
```
Comment (Ilya-Krylov) [inline mmdet/integration/nncf/compression.py:83]: Was any issue here? Why was this line changed?

**Comment by druzhkov-paul** [comment]
@LeonidBeynenson Could you please give more information on the intent of these changes?

File: mmdet/integration/nncf/compression.py (hunk 1 above)
Comment (LeonidBeynenson) [reply mmdet/integration/nncf/compression.py:83]: Yes, there was an issue, and this change fixes the bug.

**Comment by LeonidBeynenson** [comment]
2 @Ilya-Krylov , @druzhkov-paul 
The reason of the change is as follows:
1. We need to create `mmcv.Config` from a string that is read from metadata
2. At the moment the only approach to do it is to create a temporary file with the config data and then read it using `mmcv.Config.fromfile` (yes, this is _the only_ way to do it -- see the code of `mmcv.Config`)
3. But if we do not close the temporary file before `mmcv.Config.fromfile` then **sometimes** the file is empty in the moment of reading it. The reason is \*nix bufferisation
4. So we need to close the temporary file before `mmcv.Config.fromfile` 
5. This changes makes that

File: mmdet/integration/nncf/compression.py [hunk 2]
```diff
@@ -78,12 +78,24 @@ def get_nncf_config_from_meta(path):
...
+    assert isinstance(nncf_config, dict), (
+            f'Wrong nncf_config part of the config saved in the metainfo'
+            f' of the snapshot {path}:'
+            f' nncf_config={nncf_config}')
+
     nncf_config_part = {
-            'nncf_config': cfg.get('nncf_config'),
-            'find_unused_parameters': cfg.get('find_unused_parameters')
+            'nncf_config': nncf_config,
+            'find_unused_parameters': True
```
Comment (druzhkov-paul) [inline mmdet/integration/nncf/compression.py:98]: Why is it not dependent on config anymore?

Comment (LeonidBeynenson) [reply mmdet/integration/nncf/compression.py:98]: Because indeed it is a must for NNCF. This is the requirement of NNCF.
(I do not know why -- but it is a well known fact from NNCF team, that mmdetection requires this flag to make NNCF approach work)

**Comment by druzhkov-paul** [comment]
run tests

**Review by druzhkov-paul — APPROVED** [review]
